packages/compass-app-stores/src/instances-manager.ts

```diff
@@ -0,0 +1,88 @@
+import { EventEmitter } from 'events';
+import { type ConnectionInfo } from '@mongodb-js/connection-info';
+import {
+  MongoDBInstance,
+  type MongoDBInstanceProps,
+} from 'mongodb-instance-model';
+
+export enum MongoDBInstancesManagerEvents {
+  InstanceCreated = 'instance-started',
+  InstanceRemoved = 'instance-removed',
+}
+
+export type MongoDBInstancesManagerEventListeners = {
+  [MongoDBInstancesManagerEvents.InstanceCreated]: (
+    connectionInfoId: ConnectionInfo['id'],
+    instance: MongoDBInstance
+  ) => void;
+
+  [MongoDBInstancesManagerEvents.InstanceRemoved]: (
+    connectionInfoId: ConnectionInfo['id']
+  ) => void;
+};
+
+export class MongoDBInstancesManager extends EventEmitter {
+  private readonly instances = new Map<ConnectionInfo['id'], MongoDBInstance>();
+
+  createMongoDBInstance(
+    connectionInfoId: ConnectionInfo['id'],
+    instanceProps: Pick<
+      MongoDBInstanceProps,
+      '_id' | 'hostname' | 'port' | 'topologyDescription'
+    > &
+      Partial<MongoDBInstanceProps>
+  ) {
+    const instance = new MongoDBInstance(instanceProps as MongoDBInstanceProps);
+    this.instances.set(connectionInfoId, instance);
+    this.emit(
+      MongoDBInstancesManagerEvents.InstanceCreated,
+      connectionInfoId,
+      instance
+    );
+    return instance;
+  }
+
+  getMongoDBInstance(connectionInfoId: ConnectionInfo['id']) {
```

Comment:
Small suggestion: to make it clear what id is being passed here without the need to read the implementation maybe we can update the method name

```suggestion
  getMongoDBInstanceForConnection(connectionInfoId: ConnectionInfo['id']) {
```